src/lib/style-formatting/style-serializer.js

```diff
@@ -104,13 +125,19 @@ function encodeArray({ key, value, style, explode, escape }) {
     return value.map(val => valueEncoder(val)).join(`|${after}`);
   }
 
+  // If no style is present, query parameters should still **always** be encoded.
+  if (style === undefined && location === 'query') {
```

Comment:
maybe the better way is to assign the default styles and continue processing instead of having custom logic